File: quant_system/frontend/src/services/strategy.ts
/**
 * 策略管理API服务
 */
import axiosInstance from "./api";

export interface Strategy {
  id: string;
  name: string;
  description: string;
  type: string;
  parameters: Record<string, unknown>;
  status: "active" | "inactive";
  createdAt: string;
  updatedAt: string;
}

export interface StrategySignal {
  id: string;
  strategyId: string;
  symbol: string;
  type: "buy" | "sell";
  price: number;
  quantity: number;
  timestamp: string;
}

/**
 * 获取策略列表
 */
export const getStrategyList = async () => {
  return axiosInstance.get("/strategy/list");
};

/**
 * 获取策略详情
 */
export const getStrategyDetail = async (strategyId: string) => {
  return axiosInstance.get(`/strategy/${strategyId}`);
};

/**
 * 创建策略
 */
export const createStrategy = async (data: {
  name: string;
  description: string;
  strategy_type: string;
  parameters?: Record<string, unknown>;
}) => {
  return axiosInstance.post("/strategy/create", data);
};

/**
 * 更新策略
 */
export const updateStrategy = async (
  strategyId: string,
  data: {
    name?: string;
    description?: string;
    parameters?: Record<string, unknown>;
  },
) => {
  return axiosInstance.put(`/strategy/${strategyId}`, data);
};

/**
 * 删除策略
 */
export const deleteStrategy = async (strategyId: string) => {
  return axiosInstance.delete(`/strategy/${strategyId}`);
};

/**
 * 激活策略
 */
export const activateStrategy = async (strategyId: string) => {
  return axiosInstance.post(`/strategy/${strategyId}/activate`);
};

/**
 * 停用策略
 */
export const deactivateStrategy = async (strategyId: string) => {
  return axiosInstance.post(`/strategy/${strategyId}/deactivate`);
};

/**
 * 获取策略信号
 */
export const getStrategySignals = async (params: {
  strategyId: string;
  start_date?: string;
  end_date?: string;
}) => {
  return axiosInstance.get(`/strategy/${params.strategyId}/signals`, {
    params: {
      start_date: params.start_date,
      end_date: params.end_date,
    },
  });
};

/**
 * 获取策略类型列表
 */
export const getStrategyTypes = async () => {
  return axiosInstance.get("/strategy/types");
};

/**
 * 获取技术指标列表
 */
export const getTechnicalIndicators = async () => {
  return axiosInstance.get("/strategy/indicators");
};

File: quant_system/backend/src/api/routes/strategy.py
"""
策略管理路由
"""

from fastapi import APIRouter, HTTPException, Query, Body
from typing import List, Optional
from datetime import datetime
from loguru import logger

router = APIRouter()


@router.get("/list")
async def get_strategy_list():
    """获取策略列表"""
    # 这里应该从数据库查询策略列表
    return {
        "data": [],
        "total": 0
    }


@router.get("/{strategy_id}")
async def get_strategy_detail(strategy_id: str):
    """获取策略详情"""
    # 这里应该从数据库查询策略详情
    return {
        "id": strategy_id,
        "name": "",
        "description": "",
        "type": "",
        "parameters": {},
        "created_at": datetime.now().isoformat()
    }


@router.post("/create")
async def create_strategy(
    name: str = Body(..., description="策略名称"),
    description: str = Body("", description="策略描述"),
    strategy_type: str = Body(..., description="策略类型: trend/mean_reversion/arbitrage"),
    parameters: Optional[dict] = Body(None, description="策略参数")
):
    """创建新策略"""
    # 这里应该创建策略
    return {
        "id": "strategy_001",
        "name": name,
        "description": description,
        "type": strategy_type,
        "parameters": parameters or {},
        "status": "created",
        "created_at": datetime.now().isoformat()
    }


@router.put("/{strategy_id}")
async def update_strategy(
    strategy_id: str,
    name: Optional[str] = Body(None, description="策略名称"),
    description: Optional[str] = Body(None, description="策略描述"),
    parameters: Optional[dict] = Body(None, description="策略参数")
):
    """更新策略"""
    # 这里应该更新策略
    return {
        "id": strategy_id,
        "name": name,
        "description": description,
        "parameters": parameters,
        "status": "updated",
        "updated_at": datetime.now().isoformat()
    }


@router.delete("/{strategy_id}")
async def delete_strategy(strategy_id: str):
    """删除策略"""
    # 这里应该删除策略
    return {
        "id": strategy_id,
        "status": "deleted",
        "deleted_at": datetime.now().isoformat()
    }


@router.post("/{strategy_id}/activate")
async def activate_strategy(strategy_id: str):
    """激活策略"""
    # 这里应该激活策略
    return {
        "id": strategy_id,
        "status": "active",
        "activated_at": datetime.now().isoformat()
    }


@router.post("/{strategy_id}/deactivate")
async def deactivate_strategy(strategy_id: str):
    """停用策略"""
    # 这里应该停用策略
    return {
        "id": strategy_id,
        "status": "inactive",
        "deactivated_at": datetime.now().isoformat()
    }


@router.get("/{strategy_id}/signals")
async def get_strategy_signals(
    strategy_id: str,
    start_date: Optional[str] = Query(None, description="开始日期"),
    end_date: Optional[str] = Query(None, description="结束日期")
):
    """获取策略信号"""
    # 这里应该查询策略信号
    return {
        "strategy_id": strategy_id,
        "signals": [],
        "total": 0
    }


@router.get("/types")
async def get_strategy_types():
    """获取策略类型"""
    return {
        "types": [
            {
                "id": "trend",
                "name": "趋势策略",
                "description": "基于趋势跟踪的策略"
            },
            {
                "id": "mean_reversion",
                "name": "均值回归策略",
                "description": "基于均值回归的策略"
            },
            {
                "id": "arbitrage",
                "name": "套利策略",
                "description": "基于套利机会的策略"
            },
            {
                "id": "momentum",
                "name": "动量策略",
                "description": "基于动量效应的策略"
            }
        ]
    }


@router.get("/indicators")
async def get_technical_indicators():
    """获取技术指标列表"""
    return {
        "indicators": [
            {
                "id": "ma",
                "name": "移动平均线",
                "description": "简单移动平均线"
            },
            {
                "id": "ema",
                "name": "指数移动平均线",
                "description": "指数加权移动平均线"
            },
            {
                "id": "macd",
                "name": "MACD",
                "description": "移动平均收敛发散指标"
            },
            {
                "id": "rsi",
                "name": "RSI",
                "description": "相对强弱指标"
            },
            {
                "id": "boll",
                "name": "布林带",
                "description": "布林带指标"
            }
        ]
    }